Formula/newsboat.rb

```diff
@@ -14,29 +15,95 @@ class Newsboat < Formula
   uses_from_macos "libxml2"
   uses_from_macos "libxslt"
+  uses_from_macos "ncurses"
   uses_from_macos "sqlite"
 
+  # Newsboat have their own libstfl fork. Upstream libsftl is gone:
+  # https://github.com/Homebrew/homebrew-core/pull/89981
+  # They do not want to be the new upstream, but use that fork as a temporary
+  # workaround until they migrate to some rust crate
+  # https://github.com/newsboat/newsboat/issues/1881
+  resource("libstfl") do
+    url "https://github.com/newsboat/stfl.git",
+        revision: "c2c10b8a50fef613c0aacdc5d06a0fa610bf79e9"
+    sha256 "c188dcbce5e5804072bd8856e587a14d7e675ae35cb0eb715362032d7e1f9920"
+  end
+
   def install
+    resource("libstfl").stage do
+      if OS.mac?
+        ENV.append "LDLIBS", "-liconv"
+        ENV.append "LIBS", "-lncurses -lruby -liconv"
+
+        inreplace "stfl_internals.h", "ncursesw/ncurses.h", "ncurses.h"
+        inreplace %w[stfl.pc.in ruby/Makefile.snippet], "ncursesw", "ncurses"
+
+        inreplace "Makefile" do |s|
+          s.gsub! "ncursesw", "ncurses"
+          s.gsub! "-Wl,-soname,$(SONAME)", "-Wl"
+          s.gsub! "libstfl.so.$(VERSION)", "libstfl.$(VERSION).dylib"
+          s.gsub! "libstfl.so", "libstfl.dylib"
+        end
+
+        # Fix ncurses linkage for Perl bundle
+        inreplace "perl5/Makefile.PL", "-lncursesw", "-L#{MacOS.sdk_path}/usr/lib -lncurses"
+      else
+        ENV.append "LIBS", "-lncursesw -lruby"
+        inreplace "Makefile", "$(LDLIBS) $^", "$^ $(LDLIBS)"
+      end
+
+      ENV.prepend_path "PATH", Formula["python@3.10"].opt_libexec/"bin"
+      xy = Language::Python.major_minor_version "python3"
+      py_cflags = Utils.safe_popen_read("python-config", "--cflags").chomp
+
+      inreplace "python/Makefile.snippet" do |s|
+        # Install into the site-packages in the Cellar (so uninstall works)
+        s.change_make_var! "PYTHON_SITEARCH", lib/"python#{xy}/site-packages"
+        s.gsub! "gcc", "gcc #{py_cflags}"
+
+        if OS.mac?
+          s.gsub! "-lncursesw", "-lncurses -liconv"
+          s.gsub! "gcc", "gcc -undefined dynamic_lookup"
+        end
+      end
+
+      inreplace "perl5/Makefile.snippet", "perl Makefile.PL", "perl Makefile.PL INSTALL_BASE=#{libexec}"
+
+      # Fails race condition of test:
+      #   ImportError: dynamic module does not define init function (init_stfl)
+      #   make: *** [python/_stfl.so] Error 1
+      ENV.deparallelize
+      system "make"
+      system "make", "install", "prefix=#{libexec}"
+
+      (libexec/"lib").install_symlink "libstfl.so.#{version}" => "libstfl.so.#{version.major}" if OS.linux?
+    end
+
     gettext = Formula["gettext"]
 
     ENV["GETTEXT_BIN_DIR"] = gettext.opt_bin.to_s
     ENV["GETTEXT_LIB_DIR"] = gettext.lib.to_s
     ENV["GETTEXT_INCLUDE_DIR"] = gettext.include.to_s
     ENV["XML_CATALOG_FILES"] = etc/"xml/catalog"
 
+    # Remove once libsftl is not used anymore
+    ENV.prepend_path "PKG_CONFIG_PATH", libexec/"lib/pkgconfig"
```

Comment:
If we can't do static linking:
```suggestion
    ENV.prepend_path "PKG_CONFIG_PATH", libexec/"lib/pkgconfig"
    ENV.append "LDFLAGS", "-Wl,-rpath,#{libexec}/lib" if OS.linux?
```
If that doesn't work maybe `ENV.append_to_cflags` will. Or maybe dropping the `-Wl` in `LDFLAGS`. I didn't look at the `Makefile` for how they handle flags.